neutrons/RefRed
# Read/Save dead-time options to global state

# References
- EWM #[4975](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=4975)


# Description of the changes:
New class `DeadTimeSettingsModel` stores the state of the dead time settings for the Main GUI.

# Manual test for the reviewer
Before running the manual tests, install the conda environment and install the source in editable mode
```bash
> conda env create --solver libmamba --name refred-dev --file ./environment.yml
> conda activate refred-dev
(refred-dev)> pip install -e .
```
Start RefRed GUI 
```bash
(refred-dev)> PYTHONPATH=$(pwd):$PYTHONPATH ./scripts/start_refred.py
```
Then in the `Dead Time Correction` group, click in `Apply`, then `Settings` to open the settings dialog which will show default values. Change any value, then close the dialog. After this reopen the dialog to make sure the values are persistent.

# Check list for the reviewer
- [ ] Author included tests for the proposed changes
    + [ ] new functions and classes detailed docstrings, parameters documented

# Check list for the author
- [ ] I included a link to IBM EWM Story or Defect
